OpenSim/Simulation/SimbodyEngine/CustomJoint.cpp

```diff
@@ -171,20 +171,23 @@ void CustomJoint::constructCoordinates()
                 if (spatialTransform[j].hasFunction()) {
                     lf = dynamic_cast<const LinearFunction*>(
                             &spatialTransform[j].get_function() );
-                    // if displacement on axis is linear (w/ slope of 1) w.r.t.
-                    // the coordinate value, we have a pure rotation/translation
-                    if (lf && lf->getSlope() == 1.0) {
+                    // if displacement on axis is linear (w/ slope of +/-1)
+                    // w.r.t. the coordinate value, we have a pure
+                    // rotation/translation
+                    if (lf && (lf->getSlope() == 1.0 || lf->getSlope() == -1.0))
```

Comment:
I think this `if` statement should be
```cpp
if (lf && ( std::abs(lf->getSlope() - 1.) < epsilon || ... ))
```
for suitable `epsilon > 0`.

Reply:
I'm not sure about this, but I think the exact comparison is okay for a number that can be represented exactly in a floating point representation.

Reply:
That is technically correct in a closed problem, but here you are relying on the deserializer/wrapping to correctly handle precision. If `lf->getSlope()` can come directly from the deserializer, then perhaps `isEqual()` should be used instead:

https://github.com/opensim-org/opensim-core/blob/71036894bd59422d0ecfb715046317d12e3cd2ac/OpenSim/Common/Property.cpp#L39-L53

which uses an eps of 1e-7 for some reason. Otherwise, if the slope were set to 1e-8 programmatically, a different branch would be taken if the model were used immediately vs. if it were serialized/deserialized first.

Reply:
> we should also then update the value to be precisely 1 or -1.

Can we put that in CustomJoint's `extendFinalizeFromProperties()`? I think that would avoid treating the joint differently depending on whether the model has been deserialized, and then the equality test on L177 seems ok.

Reply:
Is there a reason to believe that deserialization would read a user-specified `1` or `-1` as anything but `double(1.0)` or `double(-1.0)`?

Reply:
I haven't experienced it, but I think there are models with values written like `1.00000000001` AFAICR, which likely occurred during serialization. But perhaps these weren't whole numbers.

Reply:
I think the best strategy is to go ahead with this PR. I will create a test case that demonstrates the problem.

Reply:
@aseth1 I have seen that too, with numbers that cannot be represented exactly as floating point (e.g., 0.1).

Reply:
Punted :football: to Issue #2066.

Reply:
@chrisdembia @tkuchida I am OK to merge because this fixes the larger issue of a reflected pure rotational coordinate. We can keep #2066 to remind us that serialization can be an issue here.